src/server/network.ts

```diff
@@ -212,6 +212,8 @@ export class Route {
   }
 
   async fulfill(response: { status?: number, headers?: types.HeadersArray, body?: string, isBase64?: boolean }) {
+    if (response.status && 300 <= response.status && response.status < 400)
+      throw new Error('Cannot fulfill with redirect status');
```

Comment:
nit: let's include the bad status in the error message

Reply:
Done.